### PR TITLE
refactor: update `blas/ext/base/drev` to follow current project conventions

### LINKED ISSUE #1488: [RFC]: refactor `blas/ext/base/drev` to follow current project conventions
### Description

This RFC proposes refactoring [`@stdlib/blas/ext/base/drev`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/blas/ext/base/drev) to follow current project conventions.

These conventions are outlined in more detail in https://github.com/stdlib-js/stdlib/issues/788. Namely, we want to migrate from C++ add-on interfaces to C add-on interfaces and then make various style and simplification changes.

Some example commits which perform most, if not all, of the requested changes include https://github.com/stdlib-js/stdlib/commit/d66d80cfcbab1da95cb5ae8cfae4a1fc4c5129d1 (saxpy),  https://github.com/stdlib-js/stdlib/commit/3c82fe7b6c0f9fd6ef02c86d6e2024d708ac8c14 (cswap), and https://github.com/stdlib-js/stdlib/commit/5b15c9675be434c0e11b1ec1e1f78d7b089abbcd (dasum).

### Related Issues

Related issues #788, #1152.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by performant23**
Resolves #1488.

## Description

> What is the purpose of this pull request?

This PR refactors `blas/ext/base/drev` to follow current project conventions, by updating C++ add-on interfaces to C add-on interfaces along with style and simplification changes.

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves #1488 

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Comment by performant23** [comment]
Hi, I am not able to trace through the error since it's not related to an import but a header file and it seems to be included in the code that utilizes it. If you could guide me on debugging the error, that'd be great (@Planeshifter , @kgryte , @Pranavchiku )!

Thanks!

**Comment by Pranavchiku** [comment]
Can you merge latest main to this PR and check?

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 1]
```diff
@@ -130,31 +126,15 @@ drev.ndarray( 3, x, 1, x.length-3 );
...
-    }
-    x[ i ] = sign * rand;
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100, 100 ) );
 console.log( x );
 
-drev( x.length, x, 1 );
-console.log( x );
+var y = drev( x.length, x, 1 );
+console.log( y );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
drev( x.length, x, 1 );
console.log( x );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.js [hunk 2]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var drev = require( './../lib/drev.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.native.js [hunk 3]
```diff
@@ -36,6 +36,7 @@ var drev = tryRequire( resolve( __dirname, './../lib/drev.native.js' ) );
 var opts = {
 	'skip': ( drev instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.ndarray.js [hunk 4]
```diff
@@ -21,14 +21,19 @@
...
 var isnan = require( '@stdlib/math/base/assert/is-nan' );
 var pow = require( '@stdlib/math/base/special/pow' );
-var Float64Array = require( '@stdlib/array/float64' );
 var pkg = require( './../package.json' ).name;
 var drev = require( './../lib/ndarray.js' );
 
 
+// VARIABLES //
+
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.ndarray.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.ndarray.native.js [hunk 5]
```diff
@@ -36,6 +36,7 @@ var drev = tryRequire( resolve( __dirname, './../lib/ndarray.native.js' ) );
 var opts = {
 	'skip': ( drev instanceof Error )
 };
+var rand = uniform( -100.0, 100.0 );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/benchmark/benchmark.ndarray.native.js]:
```suggestion
var rand = uniform( -10.0, 10.0 );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/examples/index.js [hunk 6]
```diff
@@ -18,28 +18,12 @@
...
-	}
-	x[ i ] = sign * rand;
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100, 100 ) );
 console.log( x );
 
-drev( x.length, x, 1 );
-console.log( x );
+var y = drev( x.length, x, 1 );
+console.log( y );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/examples/index.js]:
```suggestion
drev( x.length, x, 1 );
console.log( x );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/lib/ndarray.native.js [hunk 7]
```diff
@@ -45,10 +46,8 @@ var addon = require( './drev.native.js' );
 */
 function drev( N, x, stride, offset ) {
 	var view;
-	if ( stride < 0 ) {
-		offset += (N-1) * stride;
-	}
-	view = new Float64Array( x.buffer, x.byteOffset+(x.BYTES_PER_ELEMENT*offset), x.length-offset ); // eslint-disable-line max-len
+	offset = minViewBufferIndex( N, stride, offset );
+	view = offsetView( x, offset );
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/blas/ext/base/drev/lib/ndarray.native.js]:
```suggestion

	offset = minViewBufferIndex( N, stride, offset );

	view = offsetView( x, offset );

```

**Comment by performant23** [comment]
> Can you merge latest main to this PR and check?

Hi @Pranavchiku, you mean the develop branch right?

**Review by Pranavchiku — APPROVED** [review]
LGTM. Thanks @performant23 !

**Comment by performant23** [comment]
@Pranavchiku,  Thank you so much for your review, I have pulled the latest changes into the PR! 

cc @Planeshifter.

File: lib/node_modules/@stdlib/blas/ext/base/drev/README.md [hunk 8]
```diff
@@ -130,27 +126,11 @@ drev.ndarray( 3, x, 1, x.length-3 );
...
-    rand = round( randu()*100.0 );
-    sign = randu();
-    if ( sign < 0.5 ) {
-        sign = -1.0;
-    } else {
-        sign = 1.0;
-    }
-    x[ i ] = sign * rand;
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100, 100 ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/drev/README.md]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( -100.0, 100.0 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/docs/repl.txt [hunk 9]
```diff
@@ -84,10 +83,8 @@
 
     // Using an index offset:
     > x = new {{alias:@stdlib/array/float64}}( [ 1.0, -2.0, 3.0, -4.0, 5.0, -6.0 ] );
-    > var N = {{alias:@stdlib/math/base/special/floor}}( x.length / 2 );
-    > {{alias}}.ndarray( N, x, 2, 1 )
+    > {{alias}}.ndarray( 3, x, 2, 1 )
     <Float64Array>[ 1.0, -6.0, 3.0, -4.0, 5.0, -2.0 ]
 
     See Also
     --------
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/drev/docs/repl.txt]:
```suggestion
    --------

```

File: lib/node_modules/@stdlib/blas/ext/base/drev/examples/index.js [hunk 10]
```diff
@@ -18,27 +18,11 @@
...
-	rand = round( randu()*100.0 );
-	sign = randu();
-	if ( sign < 0.5 ) {
-		sign = -1.0;
-	} else {
-		sign = 1.0;
-	}
-	x[ i ] = sign * rand;
-}
+var x = filledarrayBy( 10, 'float64', discreteUniform( -100, 100 ) );
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/drev/examples/index.js]:
```suggestion
var x = filledarrayBy( 10, 'float64', discreteUniform( -100.0, 100.0 ) );
```

File: lib/node_modules/@stdlib/blas/ext/base/drev/src/addon.c [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/drev.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_float64array.h"
+#include <node_api.h>
+#include <assert.h>
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/drev/src/addon.c]:
```suggestion
```

`assert.h` is not explicitly used in this file.

File: lib/node_modules/@stdlib/blas/ext/base/drev/src/addon.c [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include "stdlib/blas/ext/base/drev.h"
+#include "stdlib/napi/export.h"
+#include "stdlib/napi/argv.h"
+#include "stdlib/napi/argv_int64.h"
+#include "stdlib/napi/argv_strided_float64array.h"
```
Comment (kgryte) [inline lib/node_modules/@stdlib/blas/ext/base/drev/src/addon.c:23]:
```suggestion
#include "stdlib/napi/argv_strided_float64array.h"
#include <node_api.h>
```

**Review by kgryte — APPROVED** [review]
LGTM. Thanks, @performant23 and @Pranavchiku.

**Comment by kgryte** [comment]
I am not sure the cause of the failing builds when running C examples.

**Review by kgryte — APPROVED** [review]
This should be good now. I figured out the examples error. Namely, the `manifest.json` file was not fully updated to handle the different build configurations.

**Pull request merged**